Source/Orts.Simulation/Simulation/Physics/Train.cs

```diff
@@ -9092,6 +9097,62 @@ public void UpdateExplorerMode(int signalObjectIndex)
                 }
             }
 
+            // if route does not end with signal and is too short, extend
+
+            if (!endWithSignal && totalLengthM < minCheckDistanceExplorerM)
+            {
+
+                float extendedDistanceM = minCheckDistanceExplorerM - totalLengthM;
+                TCRouteElement lastElement = newRoute[newRoute.Count - 1];
+
+                int lastSectionIndex = lastElement.TCSectionIndex;
+                TrackCircuitSection lastSection = signalRef.TrackCircuitList[lastSectionIndex];
+
+                int nextSectionIndex = lastSection.Pins[lastElement.OutPin[0], lastElement.OutPin[1]].Link;
+                int nextSectionDirection = lastSection.Pins[lastElement.OutPin[0], lastElement.OutPin[1]].Direction;
+
+                // check if last item is non-aligned switch
+
+                MisalignedSwitch[direction, 0] = -1;
+                MisalignedSwitch[direction, 1] = -1;
+
+                TrackCircuitSection nextSection = nextSectionIndex >= 0 ? signalRef.TrackCircuitList[nextSectionIndex] : null;
+                if (nextSection != null && nextSection.CircuitType == TrackCircuitSection.TrackCircuitType.Junction)
+                {
+                    if (nextSection.Pins[0, 0].Link != lastSectionIndex && nextSection.Pins[0, 1].Link != lastSectionIndex && nextSection.Pins[1, nextSection.JunctionLastRoute].Link != lastSectionIndex)
+                    {
+                        MisalignedSwitch[direction, 0] = nextSection.Index;
+                        MisalignedSwitch[direction, 1] = lastSectionIndex;
+                    }
+                }
+
+                List<int> tempSections = new List<int>();
```

Comment:
```suggestion
                List<int> tempSections;
```
This unnecessary initialization incurs a slight performance penalty.

Reply:
I didn't notice that, thanks for pointing. I have changed that in manual mode too.